tools/scripts/update-docker-tags.sh

```diff
@@ -2,7 +2,7 @@
 # Retags all docker images from latest Environments build
 # tools/scripts/update-docker-tags.sh NEW_VERSION [--release]
 
-if [ "$#" -lt 1 ] || [ "$2" != "--release" ] || [ "$#" -gt 2 ]; then
+if [ "$#" -lt 1 ] || [ ! -z "$2" -a "$2" != "--release" ] || [ "$#" -gt 2 ]; then
```

Comment:
```suggestion
if [ "$#" -lt 1 || "${2:-}" != "--release" || "$#" -gt 2 ]; then
```
That colon hyphen structure returns an empty string (or whatever is after the hyphen) if $2 is empty or unset, which makes `set -o nounset` happier. That then allows mashing it all into one set of brackets. Less noise, more easy comprehension. Feel free to disagree, though.

Reply:
I was running into an issue where "" != "--release" was actually not working properly if "$2" was an empty string. Would this solve that issue?

Reply:
If the $2 wasn't quoted, it would cause that problem because it would expand to nothing instead of an empty argument.  The "default" value can be anything, though.  Making it "${2:-x}" would cause that to be an "x" if $2 wasn't set, which would also be not equal.  But I think simply quoting the variable fixes your issue, and the extra junk makes the "don't use variables which aren't set" check behave.

Reply:
Who knew that you could do $fancy$ text in Github comments?  I guess always backtick your `$fancy$` examples, lest they be treated as LaTeX math.

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions